icu4j/main/core/src/test/resources/com/ibm/icu/dev/test/message2/test-functions.json

```diff
@@ -0,0 +1,327 @@
+{
+  "date": [
+    { "src": "{:date}", "exp": "{:date}", "errorsJs": [{ "type": "bad-input" }] },
+    {
+      "src": "{horse :date}",
+      "exp": "{|horse|}",
+      "errorsJs": [{ "name": "RangeError" }]
+    },
+    { "src": "{|2006-01-02| :date}", "exp": "1/2/06" },
+    { "src": "{|2006-01-02T15:04:06| :date}", "exp": "1/2/06" },
+    { "src": "{|2006-01-02| :date style=long}", "exp": "January 2, 2006" },
+    {
+      "src": ".local $d = {|2006-01-02| :date style=long} {{{$d :date}}}",
+      "exp": "January 2, 2006"
+    },
+    {
+      "ignore": "Can't chain :time and :date, they are differnet types",
+      "src": ".local $t = {|2006-01-02T15:04:06| :time} {{{$t :date}}}",
+      "exp": "1/2/06"
+    }
+  ],
+  "time": [
+    { "src": "{:time}", "exp": "{:time}", "errorsJs": [{ "type": "bad-input" }] },
+    {
+      "src": "{horse :time}",
+      "exp": "{|horse|}",
+      "errorsJs": [{ "name": "RangeError" }]
+    },
+    { "src": "{|2006-01-02T15:04:06| :time}", "expJs": "3:04 PM", "exp": "3:04 PM" },
+    {
+      "src": "{|2006-01-02T15:04:06| :time style=medium}",
+      "expJs": "3:04:06 PM",
+      "exp": "3:04:06 PM"
+    },
+    {
+      "src": ".local $t = {|2006-01-02T15:04:06| :time style=medium} {{{$t :time}}}",
+      "expJs": "3:04:06 PM",
+      "exp": "3:04:06 PM"
+    },
+    {
+      "ignore": "Can't chain :time and :date, they are differnet types",
+      "src": ".local $d = {|2006-01-02T15:04:06| :date} {{{$d :time}}}",
+      "exp": "3:04 PM"
+    }
+  ],
+  "datetime": [
+    {
+      "src": "{:datetime}",
+      "exp": "{:datetime}",
+      "errorsJs": [{ "type": "bad-input" }]
+    },
+    {
+      "src": "{$x :datetime}",
+      "exp": "{$x}",
+      "params": { "x": true },
+      "errors": [{ "type": "bad-input" }]
+    },
+    {
+      "ignore": "TODO",
```

Comment:
optional: remove `ignore`, and consider whether to validate against `errors` or not

Reply:
Done